media/windows/ECC-MQTT-IoT/ECC_MQTT_IoT_SQLite.py

```diff
@@ -101,16 +124,149 @@
     print(F"ECC MQTT IoT Listener application, version {eccmqtt_iot_version}, {eccmqtt_iot_date}...")
     sys.exit(0)
 
+
+class GZipRotator(logging.handlers.RotatingFileHandler):
+
+    def doRollover(self):
+
+        global done_flag
+
+        # Let the default log rollover routine do its thing first...
+        super().doRollover()
+
+        # Construct a file pattern to match all current gzip'd logs
+        base_file_pattern = self.baseFilename + ".[0-9].gz"
+        # Get a list of the current .gzip'd log files to determine the next extension to use
+        arch_files = glob.iglob(base_file_pattern)
+        # Loop through the archived log files to find the last file number in use
+        lastgz_file_num = '0'  # default value for first iteration (no current gzips)
+        for archFile in arch_files:
+            fn = archFile.split('.')
+            # The gzip'd file "number" is in array element 2
+            lastgz_file_num = fn[2]
```

Comment:
If the user foolishly put another `.` in the filename, this scheme will fail.  I don't think any of our current code does this, but someone may in the future (especially now that we have 4-5 people writing Python for ECC).  

A safer way might be to use a regular expression to search for "period some_numbers period gz end-of-line".  Also, I don't think you can assume `arch_files` will always be in sorted / numeric order, particularly if the files have differing numbers of digits.

Something like this might work (typed in Github / not tested):

```python
result = re.search('\.([0-9]+)\.gz$', archFile)
if result:
    num = int(result.group(1))
    if num > lastgz_file_num:
        lastgz_file_num = num
# ...
```

Reply:
Good points...I hadn't thought through all intricacies of this as a generic handler, but will look at your suggested changes.  In testing with < 9 backups, the list is returned in order, but I can see issues otherwise with more than 9.

Reply:
More specifically, the order that you get them back will be undefined.  It _may_ be in order, but it may not be (it kinda depends on what order they are stored in the filesystem, and that, itself, is not defined -- at least, not in Linux extX filesystems; I don't know offhand how the Windows filesystem treats ordering).

Reply:
Yes, good point, and I understand.  Will make the necessary changes.